
## Course description

NGS technologies allow researchers to sequence genomes and transcriptomes of almost any organism within a few days. However, the assembly and annotation is still very challenging and requires a deep understanding of assembly algorithms and sequencing technologies.

## Learning outcomes

Empower the students to independently de novo assemble genomes and transcriptomes 2nd and 3rd generation sequencing. Students (i) learn the theory of algorithmic assembly principles, (ii) apply it to data sets of both model & non-model organisms, (iii) assess the quality along the complete process from data generation to the genome/transcriptome assembly itself, (iv) annotate the genome by mapping the transcriptome assembly data to genome assembly data.

## Datasets

The datasets used in this course originate from:

Jiao WB, Schneeberger K. **Chromosome-level assemblies of multiple Arabidopsis genomes reveal hotspots of rearrangements with altered evolutionary dynamics.** Nature Communications. 2020;11:1–10. Available from: [http://dx.doi.org/10.1038/s41467-020-14779-y](http://dx.doi.org/10.1038/s41467-020-14779-y)

## Teachers

### Assembly part

- Rémy Bruggmann
- Heidi Tschanz-Lischer
- David M.F. Francisco
- Geert van Geest

### Annotation part

- Christian Parisod
- Manuel Poretti